aries_vcx_core/src/wallet/indy_wallet.rs

```diff
@@ -52,33 +53,45 @@ impl BaseWallet for IndySdkWallet {
         xtype: &str,
         id: &str,
         value: &str,
-        tags_json: Option<&str>,
+        tags: Option<HashMap<String, String>>,
     ) -> VcxCoreResult<()> {
+        let res = tags.map(|x| serde_json::to_string(&x)).transpose()?.to_owned();
+        let tags_json = res.as_deref();
         indy::wallet::add_wallet_record(self.wallet_handle, xtype, id, value, tags_json).await
     }
 
-    async fn get_wallet_record(&self, xtype: &str, id: &str, options_json: &str) -> VcxCoreResult<String> {
-        indy::wallet::get_wallet_record(self.wallet_handle, xtype, id, options_json).await
+    async fn get_wallet_record(&self, xtype: &str, id: &str, options: String) -> VcxCoreResult<String> {
+        let options_json = serde_json::to_string(&options)?;
```

Comment:
this is not neccessary, and converts the options into a weird string. you should be able to just pass `&options` straight into indy::waLLET::get_wallet_record